File: app/schemas/parent.py
from pydantic import BaseModel


class ParentPerformanceItem(BaseModel):
    submission_id: int
    subject: str
    chapter: str
    score: float
    error_reason: str


class ParentDashboardResponse(BaseModel):
    student_id: int
    progress: list[dict]
    recent_performance: list[ParentPerformanceItem]
    warnings: list[str]

File: eldas-frontend/src/types/parent.ts
export interface ParentPerformanceItem {
  submission_id: number;
  subject: string;
  chapter: string;
  score: number;
  error_reason: string;
}

export interface ParentDashboardResponse {
  student_id: number;
  progress: Array<{
    subject: string;
    skill_scores: {
      attempts?: number;
      averages?: Record<string, number | Record<string, number> | undefined>;
    };
    updated_at: string;
  }>;
  recent_performance: ParentPerformanceItem[];
  warnings: string[];
}
